pkg/export/export.go

```diff
@@ -765,13 +783,61 @@ func (e *Exporter) Run() error {
 		curBatch = newBatch(e.logger, opts.Efficiency.ShardCount, opts.Efficiency.BatchSize)
 	}
 
+	// Try to drain the remaining data before exiting or the time limit (15 seconds) expires.
+	// A sleep timer is added after draining the shards to ensure it has time to be sent.
+	drainShardsBeforeExiting := func() {
```

Comment:
nit: Not a must, but it might be easier to iterate on if we move `send` and `drain` to separate `Exporter` private methods 🤔